api/v1alpha4/azurecluster_default.go

```diff
@@ -148,6 +151,47 @@ func (c *AzureCluster) setAPIServerLBDefaults() {
 	}
 }
 
+func (c *AzureCluster) setNodeOutboundLBDefaults() {
+	if c.Spec.NetworkSpec.NodeOutboundLB == nil {
```

Comment:
Not sure we want this (I believe it's a change from current behavior), see my comment below in docs

Reply:
Same comment as above about having the ability to disable node outbound lb for private clusters.